packages/toolpad-app/src/toolpad/Home.tsx

```diff
@@ -264,64 +265,43 @@ function AppOpenButton({ app, activeDeployment }: AppOpenButtonProps) {
 }
 
 interface AppOptionsProps {
-  menuOpen: boolean;
-  onClick: (event: React.MouseEvent<HTMLButtonElement>) => void;
+  onRename: () => void;
+  onDelete?: () => void;
 }
 
-function AppOptions({ menuOpen, onClick }: AppOptionsProps) {
-  return (
-    <IconButton
-      aria-label="settings"
-      aria-controls={menuOpen ? 'basic-menu' : undefined}
-      aria-haspopup="true"
-      aria-expanded={menuOpen ? 'true' : undefined}
-      onClick={onClick}
-    >
-      <MoreVertIcon />
-    </IconButton>
-  );
-}
+function AppOptions({ onRename, onDelete }: AppOptionsProps) {
+  const { buttonProps, menuProps, onMenuClose } = useMenu();
 
-interface AppMenuProps {
-  menuAnchorEl: HTMLElement | null;
-  menuOpen: boolean;
-  handleMenuClose: () => void;
-  handleRenameClick: () => void;
-  handleDeleteClick: () => void;
-}
+  const handleRenameClick = React.useCallback(() => {
+    onMenuClose();
+    onRename();
+  }, [onMenuClose, onRename]);
+
+  const handleDeleteClick = React.useCallback(() => {
+    onMenuClose();
+    onDelete?.();
+  }, [onDelete, onMenuClose]);
 
-function AppMenu({
-  menuAnchorEl,
-  menuOpen,
-  handleMenuClose,
-  handleRenameClick,
-  handleDeleteClick,
-}: AppMenuProps) {
   return (
-    <Menu
-      id="basic-menu"
-      anchorEl={menuAnchorEl}
-      open={menuOpen}
-      onClose={handleMenuClose}
-      MenuListProps={{
-        'aria-labelledby': 'basic-button',
-        dense: true,
-      }}
-    >
-      {/* Using an onClick on a MenuItem causes accessibility issues, see: https://github.com/mui/material-ui/pull/30145 */}
-      <MenuItem onClick={handleRenameClick}>
-        <ListItemIcon>
-          <DriveFileRenameOutlineIcon />
-        </ListItemIcon>
-        <ListItemText>Rename</ListItemText>
-      </MenuItem>
-      <MenuItem onClick={handleDeleteClick}>
-        <ListItemIcon>
-          <DeleteIcon />
-        </ListItemIcon>
-        <ListItemText>Delete</ListItemText>
-      </MenuItem>
-    </Menu>
+    <React.Fragment>
+      <IconButton {...buttonProps} aria-label="Application menu">
+        <MoreVertIcon />
+      </IconButton>
+      <Menu {...menuProps}>
+        <MenuItem onClick={handleRenameClick}>
+          <ListItemIcon>
+            <DriveFileRenameOutlineIcon />
+          </ListItemIcon>
+          <ListItemText>Rename</ListItemText>
+        </MenuItem>
```

Comment:
Should we leave the accessibility issue link in there (assuming the issue persists of course)?

Reply:
I don't understand the comment. Isn't this how `MenuItem` is supposed to be used? If it's forcing us to do  a workaround in another part of the code then I think the comment belongs at that location.

Reply:
This was intended to be a reference to the bug report which talks about how `MenuItem` doesn't work as a button when used with a keyboard; we can remove it since we aren't doing anything specifically to tackle it